tiledb/sm/cpp_api/capi_string.h

```diff
@@ -49,17 +49,13 @@ namespace tiledb::impl {
 class CAPIString {
  public:
   CAPIString(tiledb_string_t** handle) {
```

Comment:
The constructor needs documentation.

The documentation needs to very clearly state that `CAPIString` assumes ownership of its argument and that it makes it unavailable to the caller.

Reply:
I believe it would have been more apparent with `handle` being a `tiledb_string_t*&&`, but added documentation.